app/Controllers/Http/EventsController.ts

```diff
@@ -32,21 +33,20 @@ export default class EventsController {
     const { cover } = await request.validate(CoverUploadValidator)
     event.cover = cover ? Attachment.fromFile(cover) : null
 
-    if (auth.user) {
-      // TODO set is_public if user.role equal moderator or admin
-      event.creatorEmail = auth.user.email
-      await event.save()
-    }
-
     await event.save()
     response.json({ success: true, event })
   }
 
   // update one event with id
-  public async update({ request, response }: HttpContextContract) {
+  public async update({ request, response, auth }: HttpContextContract) {
     const { id } = request.params()
     const event = await Event.findOrFail(id)
 
```

Comment:
Ich würde das ungefähr so machen:

if (!event.isPublic) //weil das ist dein pr titel
  if (!user.auth) return false // for now. user is not logged in and has no permissions, bzw. können wir hier im Frontend überlegen die Möglichkeit Events zu bearbeiten gar nicht anzuzeigen
  // user has not created this event or has no account
  if (event.creator !== user.email) return false
  if (!await Role.isAllowedToPublishContent(user.roleId)) return false

Das alles in eine Funktion im Contoller gepackt und hübsche Fehlermeldungen dazu, fertig.

Reply:
ich habe jetzt im service alle events getestet und die Funktion "isAllowedToModifyEvent" genannt, deswegen habe ich das if publich auch drin gelassen. Die verschachtelten if habe ich weg genommen und kommentiert